python/pyarrow/_flight.pyx

```diff
@@ -1150,6 +1156,38 @@ cdef class FlightClient(_Weakrefable):
                 self.client.get().Authenticate(deref(c_options),
                                                move(handler)))
 
+    def authenticateBasicToken(self, username, password,
+                               options: FlightCallOptions = None):
+        """Authenticate to the server with header token authentication.
```

Comment:
It's not really header token authentication, it's HTTP basic authentication

Reply:
Addressed in new commit.